### PR TITLE
SHOW-49: cmd/main: set a custom HTTP error handler

### PR DESCRIPTION
**Pull request by markspolakovs**
Ensure all errors get logged (and that we have control over it).

### CODE REVIEW

File: handlers/handlers.go [hunk 1]
```diff
@@ -176,10 +180,36 @@ func (h *Handlers) Start() {
...
+	var httpErr *echo.HTTPError
+	if errors.As(err, &httpErr) {
+		if strings.Contains(c.Request().Header.Get("Accept"), "application/json") {
+			c.JSON(httpErr.Code, map[string]string{"error": fmt.Sprintf("%v", httpErr.Message)})
+		} else {
+			c.String(httpErr.Code, fmt.Sprintf("%v", httpErr.Message))
+		}
+		return
+	}
+	if errors.Is(err, sql.ErrNoRows) {
```
Comment (rmil) [inline handlers/handlers.go:201]: I don't know how effective this error check will be since there are custom error's which are also "not found"

Comment (rmil) [reply handlers/handlers.go:201]: Also, shouldn't we be handling these cases? So I'd be a fan of removing this condition, but happy to chat

**Review by rmil — COMMENTED** [review]
Looks good! Definitely help with debugging live. If you could make a Jira ticket as well for it, would be pog

File: handlers/handlers.go (hunk 1 above)
Comment (markspolakovs) [reply handlers/handlers.go:201]: Yeah, that makes sense, we'd probably want a `echo.HTTPError` return for 404s - would need to look over the codebase but it makes sense to remove it.

Though on that note, there's quite a few places where we do `if err != nil { return echo.NewHTTPError(http.StatusInternalServerError, err) }` - what are your thoughts on changing those to just `return err` (or really `return fmt.Errorf("something descriptive: %w", err)` now that we have the custom handler?

Comment (rmil) [reply handlers/handlers.go:201]: > Though on that note, there's quite a few places where we do `if err != nil { return echo.NewHTTPError(http.StatusInternalServerError, err) }` - what are your thoughts on changing those to just `return err` (or really `return fmt.Errorf("something descriptive: %w", err)` now that we have the custom handler?

Agreed! (I've not done that pattern before, but it makes sense), it should definitely help us handle these pickles better

Comment (markspolakovs) [reply handlers/handlers.go:201]: It's one thing I really like about Echo - you handle the happy path, return the raw errors, and let the top-level error handler deal with them however is appropriate.

Comment (rmil) [reply handlers/handlers.go:201]: True true. I'd be a fan of setting a long term goal to refactor the code to handle these SQL errors and for the timbeing don't check and continue to serve a internal server error

Comment (markspolakovs) [reply handlers/handlers.go]: Makes sense, have filed https://ystv.atlassian.net/browse/SHOW-50 to track the tech debt.

**Review by rmil — APPROVED** [review]

**Pull request merged**